packages/framework/aqueduct/src/data-objects/pureDataObject.ts

```diff
@@ -190,6 +191,12 @@ export abstract class PureDataObject<P extends IFluidObject = object, S = undefi
         return handleFromLegacyUri<T>(`/${serviceRoutePathRoot}/${id}`, this.context.containerRuntime).get();
     }
 
+    /**
+     * Called every time the data store is initialized, before initializingFirstTime or
+     * initializingFromExisting is called.
+     */
+    protected async preInitialize(): Promise<void> { }
```

Comment:
thoughts on something like `willInitialize` instead? matches `hasInitialize`/the other method names better i think

Reply:
I feel like hasInitialize should be changed to hasInitialized :)
What do you think?

Reply:
Whoops, that's a typo on my side; it's already `hasinitialized`